File: ID/src/main/java/com/example/id/BaseApplication.java
package com.example.id;

import android.app.Application;

import com.example.id.component.AppComponent;
import com.example.id.component.DaggerAppComponent;
import com.example.id.module.AppModule;



public class BaseApplication extends Application {

    public AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        initAppInjection();
    }

    private void initAppInjection() {
        appComponent = DaggerAppComponent.builder().appModule(new AppModule(this)).build();
    }
}

File: Utils/src/main/java/com/example/utils/BaseApplication.kt
package com.example.common

import android.app.Application
import android.content.Context

/**
 * Application基类
 */
class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        context = this
//        ARouter.openLog()
//        ARouter.openDebug()
//        ARouter.init(this)
    }

    /**
     * 获取Application的上下文
     */
    companion object{
        lateinit var context: Context
    }



}